lib/request-wrapper.ts

```diff
@@ -545,3 +542,47 @@ function parseServiceErrorMessage(response: any): string | undefined {
   logger.info(`Parsing service error message: ${message}`);
   return message;
 }
+
+/**
+ * Check incoming headers to determine if the response body is meant to be JSON.
+ *
+ * @param headers - incoming headers attached to the response
+ * @returns true if content type indicates JSON, false otherwise
+ */
+function contentTypeIsJSON(headers: IncomingHttpHeaders): boolean {
+  const contentType = headers['content-type'] || '';
+  return contentType.toLowerCase() === 'application/json';
```

Comment:
I think whenever we check Content-Type for specific values, rather than doing a strict "===", we should tolerate mimetypes like `application/json; charset=utf8`.  In most situations (generator, cores, etc.) where we need to check Content-Type, we use a regex type comparison to account for the optional stuff at the end.

I did a quick check of the node core code, and I don't think we have anything like this place yet. Perhaps we could just introduce a utility function `isJsonMimetype()`.

Reply:
Yes, great call. I thought about that after I made the changes but I forgot to go back and implement it. I'm glad you pointed that out!